cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake

```diff
@@ -35,50 +35,60 @@ cmake_minimum_required(VERSION 3.5)
+#     If neither MAIN nor COMMON are specified, both are added. Specifying a
+#     source set requires a separate common output directory to be set for
+#     `apigen_generate`.
 #
 
 function(apigen_target_include_directories target)
+  set(options MAIN COMMON)
+  cmake_parse_arguments(APIGEN_TARGET_INCLUDE_DIRECTORIES "${options}" "" "" ${ARGN})
+  if(NOT APIGEN_TARGET_INCLUDE_DIRECTORIES_MAIN AND NOT APIGEN_TARGET_INCLUDE_DIRECTORIES_COMMON)
+    set(APIGEN_TARGET_INCLUDE_DIRECTORIES_MAIN TRUE)
+    set(APIGEN_TARGET_INCLUDE_DIRECTORIES_COMMON TRUE)
+  endif()
 
   get_target_property(GENERATOR ${target} APIGEN_GENERATOR)
-  get_target_property(OUTPUT_DIR ${target} APIGEN_OUTPUT_DIR)
-  get_target_property(COMMON_OUTPUT_DIR ${target} APIGEN_COMMON_OUTPUT_DIR)
-
-  # Everyone needs C++ includes
-  target_include_directories(${target}
-    PUBLIC
-      $<BUILD_INTERFACE:${OUTPUT_DIR}/cpp/include>
-      $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}/cpp/include>)
 
-  if(GENERATOR MATCHES android)
-    # Android library targets need the cpp and JNI headers to compile
+  if(APIGEN_TARGET_INCLUDE_DIRECTORIES_MAIN)
+    get_target_property(OUTPUT_DIR ${target} APIGEN_OUTPUT_DIR)
     target_include_directories(${target}
       PUBLIC
-        $<BUILD_INTERFACE:${OUTPUT_DIR}/android/jni>
-        $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}/android/jni>)
+        $<BUILD_INTERFACE:${OUTPUT_DIR}/cpp/include>
+        $<BUILD_INTERFACE:${OUTPUT_DIR}>)
 
-    # Check if we are doing a host build (no cross compilation)
-    if(NOT(${CMAKE_SYSTEM_NAME} STREQUAL "Android"))
-      find_package(JNI REQUIRED)
+    if(${GENERATOR} MATCHES android)
       target_include_directories(${target}
-        PRIVATE $<BUILD_INTERFACE:${JNI_INCLUDE_DIRS}>)
+        PUBLIC $<BUILD_INTERFACE:${OUTPUT_DIR}/android/jni>)
+    elseif(GENERATOR MATCHES dart)
+      target_include_directories(${target}
+        PUBLIC $<BUILD_INTERFACE:${OUTPUT_DIR}/dart/ffi>)
     endif()
+  endif()
 
-  elseif(GENERATOR MATCHES swift)
-    # Swift targets need the cpp and c_bridge headers to compile
+  if(APIGEN_TARGET_INCLUDE_DIRECTORIES_COMMON)
+    get_target_property(COMMON_OUTPUT_DIR ${target} APIGEN_COMMON_OUTPUT_DIR)
     target_include_directories(${target}
       PUBLIC
-        $<BUILD_INTERFACE:${OUTPUT_DIR}>
+        $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}/cpp/include>
         $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}>)
-
-  elseif(GENERATOR MATCHES dart)
-    # Dart library targets need C++ and FFI headers to compile
-    target_include_directories(${target}
-      PUBLIC $<BUILD_INTERFACE:${OUTPUT_DIR}/dart/ffi>)
-
+    if(${GENERATOR} MATCHES android)
+      target_include_directories(${target}
+        PUBLIC $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}/android/jni>)
+    endif()
   endif()
 
+  if(${GENERATOR} MATCHES android)
+    # Check if we are doing a host build (no cross compilation)
+    if(NOT(${CMAKE_SYSTEM_NAME} STREQUAL "Android"))
```

Comment:
There is variable [CMAKE_CROSSCOMPILING](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html) which you can check instead.

Reply:
I mean instead of ${CMAKE_SYSTEM_NAME} STREQUAL "Android"